src/util/transactionUtils.ts

```diff
@@ -14,8 +14,39 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
 
-import { TransactionStatusType, TransactionStatusEnum } from '../types/transactionTypes';
+import {
+  TransactionStatusType,
+  TransactionStatusEnum,
+  Payload,
+  TransactionDisplayPayload
+} from '../types/transactionTypes';
 
 export function isTransactionCompleted(transaction: TransactionStatusType): boolean {
   return transaction.status === TransactionStatusEnum.COMPLETED;
 }
+
+interface ToObject {
+  [propName: number]: number;
+}
+
+function toObject(arr: Array<any>): Object {
+  const rv: ToObject = {};
+  for (let i = 0; i < arr.length; ++i) rv[i] = arr[i];
+  return rv;
+}
+
+export function getTransactionDisplayPayload(payload: Payload | null): TransactionDisplayPayload | null {
+  const TransactionDisplayPayload = {} as TransactionDisplayPayload;
+  if (payload) {
+    const { call, origin, spec_version, weight } = payload;
+    TransactionDisplayPayload.call = toObject(call);
+    TransactionDisplayPayload.origin = {
+      SourceAccount: toObject(origin.SourceAccount)
+    };
+
+    TransactionDisplayPayload.weight = weight;
+    TransactionDisplayPayload.spec_version = spec_version;
+    return TransactionDisplayPayload;
+  }
+  return null;
+}
```

Comment:
```suggestion
export function getTransactionDisplayPayload(payload: Payload): TransactionDisplayPayload {
  const transactionDisplayPayload = {} as TransactionDisplayPayload;
  const { call, origin, spec_version, weight } = payload;
  transactionDisplayPayload.call = toObject(call);
  transactionDisplayPayload.origin = {
    SourceAccount: toObject(origin.SourceAccount)
  };

  transactionDisplayPayload.weight = weight;
  transactionDisplayPayload.spec_version = spec_version;
  return transactionDisplayPayload;
}
```

1. Keep the `null` check outside of the function (you're doing it anyway on the call site).
2. Use lower-case names for variables 😱

Reply:
1. OK
2. Daaammmnnn 😱 , that was a mistake that happened after a replace all and i didn't realize